Content.Server/GameObjects/Components/Mobs/Body/BodyTemplates/BodyTemplates.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Linq;
+using Robust.Shared.Maths;
+using System.Collections.Generic;
+using Robust.Shared.Serialization;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Interfaces.GameObjects;
+using Content.Shared.GameObjects;
+using YamlDotNet.RepresentationModel;
+using Robust.Shared.IoC;
+using Robust.Shared.Utility;
+
+namespace Content.Server.GameObjects.Components.Mobs.Body
+{
+    /// <summary>
+    ///    Core of the mobcode. It glues all the shitcode with limbs, organs 
+    ///    and body functions together with DAMAGE, making frankensteins that we call Mobs
+    /// </summary>
+
+    [Prototype("bodyTemplate")]
+    public class BodyTemplate : IPrototype, IIndexedPrototype
+    {
+#pragma warning disable CS0649
+        [Dependency]
+        protected IPrototypeManager PrototypeManager;
+#pragma warning restore
+
+        public string Name;
+        public string Id;
+        /// <summary>
+        /// The list of all mob's limbs and organs. 
+        /// </summary>
+        private List<Limb> BodyMap;
+        /// <summary>
+        /// helper list to assign children to BodyMap
+        /// </summary>
+        private List<Limb> Limbs;
+
+        /// <summary>
+        /// Entity which owns MobComponent
+        /// </summary>
+        public IEntity Owner;
+
+        /// <summary>
+        /// YAML string which iniits Blood prototype
+        /// </summary>
+        private string _bloodProt;
+
+        /// <summary>
+        /// List of YAML strings which inits all the limbs
+        /// </summary>
+        private List<string> _limbProts;
+
+        public Blood Blood; //TODO:blood should wait for reagents to get truly implemented
+
+        private Random _randomLimb;
+
+        string IIndexedPrototype.ID => Id;
+
+        void IPrototype.LoadFrom(YamlMappingNode mapping)
+        {
+            var obj = YamlObjectSerializer.NewReader(mapping);
+            obj.DataField(ref Id, "id", "");
+            obj.DataField(ref _bloodProt, "blood", "");
+            _limbProts = new List<string>();
+            foreach (var limbMap in mapping.GetNode<YamlSequenceNode>("limbs").Cast<YamlMappingNode>())
+            {
+                var limbProt = limbMap.GetNode("map").AsString();
+                _limbProts.Add(limbProt);
+            }
+        }
+
+        public void Initialize(IEntity owner)
+        {
+            Owner = owner;
+            PrototypeManager = IoCManager.Resolve<IPrototypeManager>();
```

Comment:
You don't have to explicitly do this, prototypes automatically get `[Dependency]` fields resolved.